### PR TITLE
Chore - refine VectorSearch

### PR DESCRIPTION
**Pull request by jaigouk**
changes

- moved common methods into lib/boxcars/vector_store.rb 
- Boxcars::VectorStore::Document has content, embedding and metadata
- use Boxcars::VectorStore::Document for building vectore stores and search results
- added Boxcars::VectorStore::InMemory::BuildFromFiles
- removed lib/boxcars/vector_store/similarity_search.rb use VectorSearch

why

- need to hold documents with `new` method and `call` for search multiple times with different query
- consistent structure for search classes
- ~~create query vector within SimilaritySearch~~
- revmoed SimilaritySearch and updated VectorSearch
- resolves part of https://github.com/BoxcarsAI/boxcars/issues/60

### CODE REVIEW

**Comment by jaigouk** [comment]
we have `VectorSearch` class. do we need to keep `Boxcars::VectorStore::SimilaritySearch`?

**Comment by jaigouk** [comment]
extra clean up would be adding the metadata for hnswlib search. 
or use the same structure as in_memory search.
I am not sure which one is better.

**Comment by francis** [comment]
@jaigouk - we probably don't need Boxcars::VectorStore::SimilaritySearch now

**Review by francis — APPROVED** [review]
👍

**Comment by francis** [comment]
I'm not going to merge until you give the go ahead @jaigouk

**Comment by jaigouk** [comment]
I updated gem version number to 0.2.12

**Comment by francis** [comment]
I’ll take a look.

File: lib/boxcars/vector_store/hnswlib/search.rb [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+          raw_results = search_index.search_knn(query_vector, num_neighbors)
+          raw_results.map { |doc_id, distance| lookup_embedding(doc_id, distance) }.compact
+        rescue StandardError => e
+          raise_argument_error("Error searching for #{query_vector}: #{e.message}")
+        end
+
+        def lookup_embedding(doc_id, distance)
+          return unless vector_store[doc_id]
+
+          { document: vector_store[doc_id], distance: distance }
```
Comment (jaigouk) [inline lib/boxcars/vector_store/hnswlib/search.rb:65]: should we return array of `Boxcars::VectorStore::Document` instances?

File: spec/boxcars/vector_search_spec.rb [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+      let(:hnswlib_index) { 'spec/fixtures/embeddings/test_hnsw_index.bin' }
+
+      before do
+        allow(Boxcars::VectorStore::EmbedViaOpenAI).to receive(:call).and_return(query_vector)
+      end
+
+      it 'returns an Document array' do
+        expect(search_result).to be_a(Array)
+
+        expect(search_result.first[:document]).to be_a(Boxcars::VectorStore::Document)
```
Comment (jaigouk) [inline spec/boxcars/vector_search_spec.rb:46]: we return `Boxcars::VectorStore::Document`

**Pull request merged**